### PR TITLE
Angular table and JSON file

### PR DESCRIPTION
**Pull request by PatrykDawczak**

### CODE REVIEW

File: app/index.html [hunk 1]
```diff
@@ -3,29 +3,36 @@
 <head>
   <meta charset="utf-8">
   <meta http-equiv="X-UA-Compatible" content="IE=edge">
-  <title>My Rest App</title>
-  <meta name="description" content="">
+  <title>Menu App</title>
+  <meta name="description" content="Aplikacja Menu dla sprawnego obsługiwania zamówień w restauracji!">
```
Comment (TomaszGabrysiak) [inline app/index.html]: I think it's not used by any search engine so can be omitted.

File: app/app.js [hunk 2]
```diff
@@ -12,25 +11,28 @@ var app = angular.module('app', ['ui.router']);
...
-            "name": "Drink 3",
-            "description": "Lorem Ipsum jest tekstem stosowanym jako przykładowy wypełniacz w przemyśle poligraficznym",
-            "img": "waiter/img/drink_3.jpg",
-            "price": "6"
-        }];
-    });
-    
\ No newline at end of file
+    app.controller('DishCtrl', function ($scope, $http) {
+     $http.get('json/drink.json').success(function(data) {
```
Comment (TomaszGabrysiak) [inline app/app.js]: First of all - all code related with downloading data should be placed in services. Second thing  - this way of chaining requests is really strange. Please use promises and function 'then' to chain requests.

Each service should expose functions named 'getXxx()', eg. 'getDrinks' which will return promise. Then you can chain them in a right way in controllers.

File: app/cook/cook.html [hunk 3]
```diff
@@ -1,3 +1,4 @@
 <div class="container">
-    <p>Lista zamówień - widok kucharza</p>
+    <h3>Widok kucharza!</h3>
```
Comment (TomaszGabrysiak) [inline app/cook/cook.html]: No .row in container

File: app/index.html [hunk 4]
```diff
@@ -3,29 +3,36 @@
...
   <meta name="viewport" content="width=device-width, initial-scale=1">
-  <link rel="stylesheet" href="bower_components/html5-boilerplate/dist/css/normalize.css">
-  <link rel="stylesheet" href="bower_components/html5-boilerplate/dist/css/main.css">
   <link rel="stylesheet" href="app.css">
```
Comment (TomaszGabrysiak) [inline app/index.html]: Custom css file should be loaded after all other styles

File: app/index.html [hunk 5]
```diff
@@ -3,29 +3,36 @@
...
-  <link rel="stylesheet" href="bower_components/bootstrap/dist/css/bootstrap.css">
-  <script src="bower_components/html5-boilerplate/dist/js/vendor/modernizr-2.8.3.min.js"></script>
+  <link rel="stylesheet" href="bower_components/bootstrap/dist/css/bootstrap.min.css">
+  <script src="bower_components/jquery/dist/jquery.min.js"></script>
+  <script src="bower_components/bootstrap/dist/js/bootstrap.min.js"></script>
+  <script src="bower_components/bootstrap/js/collapse.js"></script>
```
Comment (TomaszGabrysiak) [inline app/index.html]: I think that 'bootstrap.min.js' contain collapse as well as all of the others components.

File: app/index.html [hunk 6]
```diff
@@ -3,29 +3,36 @@
...
 </head>
 <body>
   <div class="container">
```
Comment (TomaszGabrysiak) [inline app/index.html]: .row and .col is missing

File: app/index.html [hunk 7]
```diff
@@ -3,29 +3,36 @@
...
   </div>
   <div ui-view></div>
   <script src="bower_components/angular/angular.js"></script>
   <script src="bower_components/angular-ui-router/release/angular-ui-router.js"></script>
+  <script src="bower_components/html5-boilerplate/dist/js/vendor/modernizr-2.8.3.min.js"></script>
   <script src="app.js"></script>
   <script src="waiter/waiter.js"></script>
   <script src="cook/cook.js"></script>
-  <script src="waiter/countries.json"></script>
+  <script src="json/drink.json"></script>
```
Comment (TomaszGabrysiak) [inline app/index.html]: Why do you load json data here? It's not necessary

File: app/waiter/waiter.html [hunk 8]
```diff
@@ -1,99 +1,163 @@
...
-        <li><a class="btn btn-default" href="#" role="button">Napoje</a></li>
-        <li><a class="btn btn-default" href="#" role="button">Przystawki</a></li>
-        <li><a class="btn btn-default" href="#" role="button">Zupy</a></li>
-        <li><a class="btn btn-default" href="#" role="button">Danie główne</a></li>
-        <li><a class="btn btn-default" href="#" role="button">Dodatki</a></li>
-        <li><a class="btn btn-default" href="#" role="button">Deser</a></li>
-        <li><a class="btn btn-default" href="#" role="button">Podsumowanie</a></li>
-    </ul>
+    <div class="btn-group btn-group-justified">
+        <a href="#home" data-toggle="collapse" class="btn btn-default" role="button">Home</a>
```
Comment (TomaszGabrysiak) [inline app/waiter/waiter.html]: Please use reverse url matching using 'ui-sref' directive. It comes with ui.router.

File: app/waiter/waiter.html [hunk 9]
```diff
@@ -1,99 +1,163 @@
...
-                <option value="3">3</option>
-                <option value="4">4</option>
-                <option value="5">5</option>
-            </select>
-        </div>
+    <div id="home" class="collapse in" ng-controller="DishCtrl">
+        <form role="form">
+            <div class="form-group">
+                <h4>Wybierz stolik:</h4>
+                <select class="form-control" ng-model="selectedTable">
```
Comment (TomaszGabrysiak) [inline app/waiter/waiter.html]: Please use `ng-options` directive instead of placing options in `ng-repeat`

File: app/waiter/waiter.html [hunk 10]
```diff
@@ -1,99 +1,163 @@
...
+                <h4>Liczna gości:</h4>
+                <select class="form-control" ng-model="selectedPeople">
+                    <option ng-repeat="x in table" value="{{x.peopleID}}">{{x.peopleID}}</option>
+                </select>
+            </div>
+            <h5>Przyjmij zamówienie od {{selectedPeople}} gości!</h5>
+        </form>
     </div>
 </div>
 <div class="container">
```
Comment (TomaszGabrysiak) [inline app/waiter/waiter.html]: As nearly everywhere - the grid is invalid. Please use `.row` and `.col` classes

File: app/waiter/waiter.html [hunk 11]
```diff
@@ -1,99 +1,163 @@
...
-    <div class="row">
-        <div class="col-md-12">
-            <h1>Danie główne</h1>
+    <div id="dish3" class="collapse" ng-controller="DishCtrl">
+        <div class="table-responsive">
+            <table class="table">
+                <tr ng-repeat="x in soup">
+                    <td>{{ x.name }}</td>
```
Comment (TomaszGabrysiak) [inline app/waiter/waiter.html]: Try using more informative names for iterators. `x` isn't good option

File: app/waiter/waiter.html [hunk 12]
```diff
@@ -1,99 +1,163 @@
...
+                    <td><button type="button" class="btn btn-success">Zamów</button></td>
+                </tr>
+            </table>
         </div>
     </div>
 </div>
-
-
+<div class="container">
+    <div id="end" class="collapse">
```
Comment (TomaszGabrysiak) [inline app/waiter/waiter.html]: I would recommend adding prefixes to the custom ids/classes. It helps you to distinguish your custom classes and other classes provided by libraries etc.

File: app/index.html (hunk 1 above)
Comment (PatrykDawczak) [inline app/index.html]: Done

File: app/cook/cook.html (hunk 3 above)
Comment (PatrykDawczak) [inline app/cook/cook.html]: done

File: app/index.html (hunk 4 above)
Comment (PatrykDawczak) [inline app/index.html]: done

File: app/index.html (hunk 5 above)
Comment (PatrykDawczak) [inline app/index.html]: done

File: app/index.html (hunk 6 above)
Comment (PatrykDawczak) [inline app/index.html]: done

File: app/index.html (hunk 7 above)
Comment (PatrykDawczak) [inline app/index.html]: done

File: app/waiter/waiter.html (hunk 9 above)
Comment (PatrykDawczak) [inline app/waiter/waiter.html]: done

File: app/waiter/waiter.html (hunk 10 above)
Comment (PatrykDawczak) [inline app/waiter/waiter.html]: done

File: app/waiter/waiter.html (hunk 11 above)
Comment (PatrykDawczak) [inline app/waiter/waiter.html]: done

File: app/app.js [hunk 13]
```diff
@@ -12,25 +11,15 @@ var app = angular.module('app', ['ui.router']);
...
-            "description": "Lorem Ipsum jest tekstem stosowanym jako przykładowy wypełniacz w przemyśle poligraficznym",
-            "img": "waiter/img/drink_3.jpg",
-            "price": "6"
-        }];
+    app.controller('menuCtrl', function($scope, $http) {
+    $http.get("json/menu.json").then(function (response) {
+        $scope.getMenu = response.data.menu;
+        $scope.getTable = response.data.table;
+        $scope.getPeople = response.data.people;
+        $scope.getDrink = response.data.drink;
```
Comment (TomaszGabrysiak) [inline app/app.js]: Hmm, I would prefer to name these variables like `$scope.drinks, $scope.tables, $scope.people etc`. Name `getXxx` suggests this is a function returning sth.

File: app/app.js [hunk 14]
```diff
@@ -12,25 +11,15 @@ var app = angular.module('app', ['ui.router']);
...
+        $scope.getAppetizer = response.data.appetizer;
     });
-    
\ No newline at end of file
+        $scope.showMe = false;
+        $scope.myFunc = function() {
```
Comment (TomaszGabrysiak) [inline app/app.js]: rename

File: app/app.js [hunk 15]
```diff
@@ -12,25 +11,15 @@ var app = angular.module('app', ['ui.router']);
...
+    $http.get("json/menu.json").then(function (response) {
+        $scope.getMenu = response.data.menu;
+        $scope.getTable = response.data.table;
+        $scope.getPeople = response.data.people;
+        $scope.getDrink = response.data.drink;
+        $scope.getAppetizer = response.data.appetizer;
     });
-    
\ No newline at end of file
+        $scope.showMe = false;
```
Comment (TomaszGabrysiak) [inline app/app.js]: rename

File: app/cook/cook.html [hunk 16]
```diff
@@ -1,3 +1,7 @@
 <div class="container">
-    <p>Lista zamówień - widok kucharza</p>
+    <div class="row">
+        <div class="col-md-12">
```
Comment (TomaszGabrysiak) [inline app/cook/cook.html]: What about the smallest devices? Please read about the `mobile first` conception used in bootstrap.

File: app/cook/cook.js [hunk 17]
```diff
@@ -0,0 +1,41 @@
+app.controller('CrudCtrl', ['$scope',
+    function($scope) {
+        $scope.Profiles = [
```
Comment (TomaszGabrysiak) [inline app/cook/cook.js]: profiles

File: app/app.js [hunk 18]
```diff
@@ -1,36 +1,50 @@
 var app = angular.module('app', ['ui.router']);
     app.config(['$urlRouterProvider', '$stateProvider', function($urlRouterProvider, $stateProvider) {
       $urlRouterProvider.otherwise('/');
-
       $stateProvider
           .state('waiter', {
             url: '/waiter',
             templateUrl: 'waiter/waiter.html'
           })
+          .state('waiter.dish1', {
```
Comment (TomaszGabrysiak) [inline app/app.js]: `dishX` seems to be not a good name here

**Pull request merged**